lightning/src/routing/router.rs

```diff
@@ -216,84 +359,200 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 								cltv_expiry_delta: 0,
 							},
-						)
+							src_lowest_inbound_fees: RoutingFees {
+								base_msat: fee_base_msat,
+								proportional_millionths: fee_proportional_millionths,
+							},
+							channel_fees: $directional_info.fees,
+							next_hops_fee_msat: u64::max_value(),
+							hop_use_fee_msat: u64::max_value(),
+							total_fee_msat: u64::max_value(),
+						}
 					});
+
+					let mut hop_use_fee_msat = 0;
+					let mut total_fee_msat = $next_hops_fee_msat;
+
+					// Ignore hop_use_fee_msat for channel-from-us as we assume all channels-from-us
+					// will have the same effective-fee
 					if $src_node_id != *our_node_id {
-						// Ignore new_fee for channel-from-us as we assume all channels-from-us
-						// will have the same effective-fee
-						total_fee += new_fee;
-						if let Some(fee_inc) = final_value_msat.checked_add(total_fee).and_then(|inc| { (old_entry.2 as u64).checked_mul(inc) }) {
-							total_fee += fee_inc / 1000000 + (old_entry.1 as u64);
-						} else {
-							// max_value means we'll always fail the old_entry.0 > total_fee check
-							total_fee = u64::max_value();
+						match compute_fees(amount_to_transfer_over_msat, $directional_info.fees) {
+							// max_value means we'll always fail
+							// the old_entry.total_fee_msat > total_fee_msat check
+							None => total_fee_msat = u64::max_value(),
+							Some(fee_msat) => {
+								hop_use_fee_msat = fee_msat;
+								total_fee_msat += hop_use_fee_msat;
+								if let Some(prev_hop_fee_msat) = compute_fees(total_fee_msat + amount_to_transfer_over_msat,
+																			old_entry.src_lowest_inbound_fees) {
+									if let Some(incremented_total_fee_msat) = total_fee_msat.checked_add(prev_hop_fee_msat) {
+										total_fee_msat = incremented_total_fee_msat;
+									}
+									else {
+										// max_value means we'll always fail
+										// the old_entry.total_fee_msat > total_fee_msat check
+										total_fee_msat = u64::max_value();
+									}
+								} else {
+									// max_value means we'll always fail
+									// the old_entry.total_fee_msat > total_fee_msat check
+									total_fee_msat = u64::max_value();
+								}
+							}
 						}
 					}
+
 					let new_graph_node = RouteGraphNode {
 						pubkey: $src_node_id,
-						lowest_fee_to_peer_through_node: total_fee,
-						lowest_fee_to_node: $starting_fee_msat as u64 + new_fee,
+						lowest_fee_to_peer_through_node: total_fee_msat,
+						lowest_fee_to_node: $next_hops_fee_msat as u64 + hop_use_fee_msat,
+						value_contribution_msat: value_contribution_msat,
 					};
-					if old_entry.0 > total_fee {
+
+					// Update the way of reaching $src_node_id with the given $chan_id (from $dest_node_id),
+					// if this way is cheaper than the already known
+					// (considering the cost to "reach" this channel from the route destination,
+					// the cost of using this channel,
+					// and the cost of routing to the source node of this channel).
```

Comment:
I don't understand this comment, what `total_fee_msat` is representing here ?

It sounds to weight `total_fee_msat` as the whole fees paid across the payment path. However, `total_fee_msat` is assigned on the sum of `$next_hops_fee_msat` and `hop_use_fee_msat`, but it doesn't include the fee paid to go from this chan to payment path source (the payer). So what do you mean by "the cost of routing to the source node of this channel" ?

I might be confused by the fact we're traversing payee-to-payer.

Reply:
I hoped you'd refer to the comment where this field is defined:
```
	/// Used to compare channels when choosing the for routing.
	/// Includes paying for the use of a hop and the following hops, as well as
	/// an estimated cost of reaching this hop.
	/// Might get stale when fees are recomputed. Primarily for internal use.
```

Is it not clear enough?

Reply:
As this step of the routing, `total_fee_msat` is the sum of `$next_hops_fee_msat` (L513),
`hop_use_fee_msat` (L524) and potentially of `incremented_total_fee_msat` (L528). What term corresponds to "estimated cost of reaching this hop" here ?

Reply:
``Some(incremented_total_fee_msat) = total_fee_msat.checked_add(prev_hop_fee_msat)``

So, you see how it includes `prev_hop_fee_msat`

Reply:
Right, but let's say you have the source node S of this hop N, `prev_hop_fee_msat` is the fee used to reach S paid on N-1 not the fees aggregation of all channel N-1 to N-i, where _i_ would be the shortest-path from payer to reach S. Because actually we don't know _i_ yet at this step of routing ?

Reply:
Ok, so the issue is that it's not really `total`, but it's some lower bound? How we should call it then?
How about `lower_bound_aggregate_fee_msat`, and explain why it's lower bound?